recipes/tree-sitter/all/conanfile.py

```diff
@@ -0,0 +1,65 @@
+import os
+from conans import ConanFile, AutoToolsBuildEnvironment, tools
+from conans.errors import ConanInvalidConfiguration
+import shutil
+
+required_conan_version = ">=1.29.1"
+
+class TreeSitterConan(ConanFile):
+    name = "tree-sitter"
+    description = "Tree-sitter is a parser generator tool and an incremental parsing library. It can build a concrete syntax tree for a source file and efficiently update the syntax tree as the source file is edited."
+    topics = ("parser")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://tree-sitter.github.io/tree-sitter"
+    license = "MIT"
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"fPIC": [True, False],
+               "shared": [True, False]}
+    default_options = {"fPIC": True,
+                       "shared": False}
+    _autotools = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def requirements(self):
+        if self.settings.os == "Windows":
+            raise ConanInvalidConfiguration("tree-sitter is not support on {}.".format(self.settings.os))
```

Comment:
On mobile but this looks like it's in the `requirements` I believe this should go in `configures`

Reply:
As I understand after[reading](https://docs.conan.io/en/latest/reference/conanfile/methods.html#configure-config-options) of this article `configure` section would be called before `requirements`, so replacement of function is just performance optimization, am I right or something missed in my thoughts?

Reply:
I am certainly no expert on the subject, my understanding is there are various workflows, example CCI will calculate the `package_id` it does not need the requires to be triggered (SSE4 posted a list in another issue) so there is some level of correctness too